packages/expo-router/src/link/preview/native.tsx

```diff
@@ -1,7 +1,11 @@
 'use client';
 
 import { requireNativeView } from 'expo';
-import { Platform, StyleSheet, type ViewProps } from 'react-native';
+import Constants from 'expo-constants';
+import { StyleSheet, type ViewProps } from 'react-native';
+
+const areNativeViewsAvailable =
+  process.env.EXPO_OS === 'ios' && Constants?.expoConfig?.newArchEnabled !== false;
```

Comment:
```suggestion
  process.env.EXPO_OS === 'ios' && globalThis.RN$Bridgeless === true;
```

the `newArchEnabled` from app.json might not be correct. for example:
- people have `newArchEnabled=false` and `.ios.newArchEnabled=true`.
- people manually run `RCT_NEW_ARCH_ENABLED=0 pod install`.

i would recommend to check the runtime flag from react-native. even though bridgless is not 100% equal to new arch. but most of time it is.